posts/jenkins-vs-aws-codepipeline.md

```diff
@@ -0,0 +1,49 @@
+
+If you have worked with CI/CD, chances are you have used Jenkins, GitLab, or GitHub Actions, but one service I don’t often hear folks talk about is AWS CodePipeline. In this blog, we will make some comparisons between Jenkins and AWS CodePipeline, and see if maybe it’s time for your projects to make the switch.
+
+
+# What is Jenkins?
+
+Jenkins seems like the place to start. As of writing this blog, Jenkins is by far the market leader for CI/CD pipelines, and has the added benefit of being open source. Boasting over 1,800 plugins, Jenkins has been able to integrate with so many different tools and services with little additional work. Jenkins also easily integrates with all of the major public cloud providers, and many developer tools. Jenkins’ other main benefit is its price, free. While there are companies that offer Jenkins as a SaaS solution, anyone can download and start using Jenkins. This of course comes with the downside of needing to maintain everything about your Jenkins environment, which at scale can become difficult.
+
+
+# What is AWS CodePipeline?
+
+AWS CodePipeline is a fully managed CI/CD service, and is actually the orchestration of a few other AWS services including CodeCommit, CodeBuild, and CodeDeploy. Because it is an AWS service, you can use CodePipeline to deploy to other servers running the agent. Since these are all managed services, there is no underlying infrastructure you will have to maintain. But managed services come with a cost. There is a cost for having an active pipeline, as well as the compute time required for building and some deployment operations. 
+
+
+# Price comparison
+
+We mentioned earlier that Jenkins is free and open source and that there are costs associated with aspects of CodePipeline, but there is more to the story than that. While Jenkins is free to run, you must consider both the cost of the infrastructure it is going to run on, as well as the cost of time and knowledge to maintain Jenkins, the infrastructure, and any plugins. CodePipeline on the other hand has no cost associated with infrastructure, but has a compute cost every time you kick off a build. These prices depend on the instance type you need to build on and are billed by the minute. The full price breakdown can be found [here in the AWS docs](https://aws.amazon.com/codebuild/pricing/?nc=sn&loc=3). 
+
+
+# Scalability
+
+When using CodePipeline, each pipeline run executes independently and in parallel with each other. Since CodePipeline is a managed service, you can run many pipelines in parallel without having to deal with additional infrastructure. Jenkins on the other hand is limited by the number of executors you have. While there are ways to have the Jenkins executor be a docker container that spins up on demand, you will still be limited by your hardware. Jenkins also only allows a single instance of a pipeline to be running at once, so you must wait until your deployment is complete before starting your next build.
```

Comment:
```suggestion
When using CodePipeline, each pipeline run executes independently and in parallel with each other. Since CodePipeline is a managed service, you can run many pipelines in parallel without having to deal with additional infrastructure. Jenkins on the other hand is limited by the number of executors you have. While there are ways to have the Jenkins executor be a Docker container that spins up on demand, you will still be limited by your hardware. Jenkins also only allows a single instance of a pipeline to be running at once, so you must wait until your deployment is complete before starting your next build.
```